loleaflet/Makefile.am

```diff
@@ -519,13 +519,12 @@ node_modules: package.json archived-packages
 	@npm install
 	@touch node_modules
 
-Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status node_modules
-	@case '$?' in \
-	 *config.status*) \
-	  cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh;; \
-	 *) \
-	  cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@;; \
-	esac;
+define target_makefile
+Makefile: $$(srcdir)/Makefile.in $$(top_builddir)/config.status $(1)
+	  @cd $$(top_builddir) && $$(MAKE) $$(AM_MAKEFLAGS) am--refresh
+endef
+
+$(eval $(call target_makefile,$(if $(findstring clean,$(MAKECMDGOALS)),,node_modules)))
```

Comment:
Clearly your 'make' skills are beyond what most people get by. But explaining in a comment what we're doing is important, because other people are not expected to know our intentions from simply reading advanced code.